### PR TITLE
na_ontap_volume - added qos policy module options

### PR DESCRIPTION
**Pull request by vicmunoz**
##### SUMMARY
Added new options:
-- qos_policy_group
-- qos_adaptive_policy_group

##### ISSUE TYPE
Feature Pull Request

##### COMPONENT NAME
na_ontap_volume

### CODE REVIEW

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_volume.py]: I don't see the point of adding these intermediate variables.

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_volume.py]: I don't object to the reformatting, but is it really necessary?

Are you using some formatting tool?

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_volume.py]: why change this and below?

**Review by carchi8py — CHANGES_REQUESTED** [review]
Can we limit the change to just adding the qos policy necessary changes?

**Comment by vicmunoz** [comment]
Other changes added are just pylint recommendations or similar. Is it ok for you or prefer to remove them?

**Comment by carchi8py** [comment]
Which one are pylint recommendations?

and are you running the standard pylint or the modify one ansible uses
ansible-test sanity --docker=default --test pylint

**Comment by vicmunoz** [comment]
alignments, lines too long... 
I've executed pylint standard. I didn't know ansible pylint. 
I'll review and check using ansible sanity test from now on

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (vicmunoz) [reply lib/ansible/modules/storage/netapp/na_ontap_volume.py]: Apols, I was using python standard formatter that recommends line length of 100 instead of 160 that recommends ansible. I will revert

Comment (vicmunoz) [reply lib/ansible/modules/storage/netapp/na_ontap_volume.py]: idem, to avoid 100 chars line length, reverting in my next commit

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (vicmunoz) [reply lib/ansible/modules/storage/netapp/na_ontap_volume.py]: reverting

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (vicmunoz) [reply lib/ansible/modules/storage/netapp/na_ontap_volume.py]: reverting

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_volume.py]: why do you need this variable?

**Review by lonico — APPROVED** [review]
shipit

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (vicmunoz) [reply lib/ansible/modules/storage/netapp/na_ontap_volume.py]: you're right, is not needed. thanks,

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (thedoubl3j) [inline lib/ansible/modules/storage/netapp/na_ontap_volume.py]: is this a bool or permission? If it is, perfect but if not I would say `Specifies` rather than `Allows`.

Comment (thedoubl3j) [reply lib/ansible/modules/storage/netapp/na_ontap_volume.py]: could you possibly add examples in as well for docs?

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (thedoubl3j) [inline lib/ansible/modules/storage/netapp/na_ontap_volume.py:231]: same as the comments above, mainly the examples one that we would like added if possible :)

**Review by thedoubl3j — CHANGES_REQUESTED** [review]
if we could get a couple of examples of the new features added in, that would be great

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (vicmunoz) [reply lib/ansible/modules/storage/netapp/na_ontap_volume.py]: updated to 'Specifies' and included in example

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (vicmunoz) [reply lib/ansible/modules/storage/netapp/na_ontap_volume.py:231]: updated to 'Specifies'

**Comment by vicmunoz** [comment]
updated details and examples

**Review by thedoubl3j — APPROVED** [review]

**Comment by thedoubl3j** [comment]
@carchi8py @lonico please re ship it when you get a moment to review.

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_volume.py:251]: can you also add qos_adaptive_policy_group as an example  (here or some new example) ?

**Review by lonico — APPROVED** [review]
shipit

File: lib/ansible/modules/storage/netapp/na_ontap_volume.py
Comment (vicmunoz) [reply lib/ansible/modules/storage/netapp/na_ontap_volume.py:251]: added a new example with adaptive qos policy

**Review by lonico — APPROVED** [review]
Shipit

**Review by thedoubl3j — APPROVED** [review]
examples added for both, lgtm.

**Comment by thedoubl3j** [comment]
shipit

**Comment by thedoubl3j** [comment]
confirmed by vendor maintainers as good to go. @lonico @carchi8py merging

**Pull request merged**
